sessions-to-ics.js

```diff
@@ -4,98 +4,116 @@ const ics = require("ics");
+
+    const startDate = vegasDate(month, date, startHour, startAp, startMin);
+    const endDate = vegasDate(month, date, endHour, endAp, endMin);
+
+    const duration = endDate.diff(startDate, ['hours', 'minutes']);
+    return {
+        start: [startDate.year, startDate.month, startDate.day, startDate.hour, startDate.minute],
+        duration: { hours: duration.hours, minutes: duration.minutes },
+    };
 }
 
-function toIcs(row) {
-  let timeDetails = {};
-  try {
-    timeDetails = extractTimeDetails($(".sessionTimeList", row).html());
-  } catch (err) {
-    console.log(err);
-    console.log($(row).html());
-    return undefined;
-  }
-  const type = $(".type", row).text();
-  let description = $(".abstract", row).html();
-  description = description.substring(0, description.indexOf(" <a"));
-
-  const event = {
-    start: timeDetails.start,
-    startInputType: "utc",
-    duration: timeDetails.duration,
-    location: $(".sessionRoom", row).text().replace(/^[^A-Za-z]*/g, ""),
-    title: $(".openInPopup span", row).text(),
-    description: `${type}\n\n${description}`,
-  };
-  return event;
+function extractTitle(sessionContainer) {
+    const titleAndCode = $(".awsui-util-mt-m > div", sessionContainer);
+    const title = $(titleAndCode[0]).text();
+    const code = $(titleAndCode[1]).text();
+    return `[${code}] ${title}`;
 }
 
-function toNormalizedType(row) {
-  const type = $(".type", row).text();
-  return type.toLowerCase().replace(/ +/, '-');
+function extractDescription(sessionContainer) {
+    return $('.sanitized-html', sessionContainer).html();
+}
+
+function extractSessionProps(sessionContainer) {
+    const sessionInfoElements = $('.awsui-util-mr-xs', sessionContainer);
+    const sessionInfo = {};
+    sessionInfoElements.each((idx, n) => { sessionInfo[$(n.parent.children[0]).text().replace(":", "")] = $(n.parent.children[1]).text() });
+    return sessionInfo;
+}
+
+function toIcs(sessionContainer) {
+    const sessionProps = extractSessionProps(sessionContainer)
+    let timeDetails = {};
+    try {
+        timeDetails = extractTimeDetails(sessionProps);
+    } catch (err) {
+        console.log(err);
+        console.log($(sessionContainer).html());
+        return undefined;
+    }
+    const sessionType = sessionProps["Session type"];
+    const description = extractDescription(sessionContainer);
+
+    const event = {
+        start: timeDetails.start,
+        startInputType: "utc",
+        duration: timeDetails.duration,
+        location: sessionProps["Location"],
+        title: extractTitle(sessionContainer),
+        description: `${sessionType}\n\n${description}`,
+        sessionType: sessionType
+    };
+    return event;
 }
 
 const events = {};
-$(".sessionRow").each((idx, row) => {
-  const event = toIcs(row);
-  if (event) {
-    const eventType = toNormalizedType(row);
-    if (!events.hasOwnProperty(eventType)) {
-      events[eventType] = []
+$(".awsui-grid").each((idx, row) => {
```

Comment:
```suggestion
$(".awsui-util-mb-xl").each((idx, row) => {
```

The `awsui-grid` contains all of the events. It looks like each favorited event is wrapped in a `.awsui-util-mb-xl` div. That gets it to recognize all of the events. The durations aren't parsing correctly yet and I wasn't able to import into Google Calendar, so need to look into that, but this is a first step!